File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription)

(def dna-to-rna {
                 \G \C,
                 \C \G,
                 \T \A,
                 \A \U
                 })

(defn convert [dna-strand]
  (if-let [rna-strand (dna-to-rna dna-strand)]
    rna-strand
    (throw (AssertionError. "unknown dna strand"))))

(defn to-rna [dna]
  (apply str (map convert dna)))

File: kotlin/rna-transcription/src/main/kotlin/RnaTranscription.kt
/**
 * Created by yannickgrenzinger on 11/08/2016.
 */
object RnaTranscription {

    private fun convertStrand(strand: Char) : Char {
        return when(strand) {
            'G'  -> 'C'
            'C'  -> 'G'
            'T'  -> 'A'
            'A'  -> 'U'
            else -> throw Exception("This DNA strand is unknown")
        }
    }

    fun  ofDna(dna: String): String {
        return dna.map { convertStrand(it) }.joinToString("")
    }
}